File: app/src/main/java/io/mgba/Data/Remote/Interfaces/IRequest.java
package io.mgba.Data.Remote.Interfaces;

import io.mgba.Data.Remote.DTOs.GameJSON;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface IRequest {

    String BASE_URL = "https://andr3carvalh0.github.io/Databases/mGBA/";

    @GET("Games_{lang}/{md5}.json")
    Call<GameJSON> getGameInformation(@Path("md5") String md5, @Path("lang") String lang);
}

File: app/src/main/java/io/mgba/data/remote/interfaces/IRequest.kt
package io.mgba.data.remote.interfaces

import io.mgba.data.remote.RetrofitWrapper
import io.mgba.data.remote.dtos.GameJSON
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface IRequest {

    @GET("Games_{lang}/{md5}.json")
    fun getGameInformation(@Path("md5") md5: String, @Path("lang") lang: String): Call<GameJSON>
}

fun getGamesService(): IRequest {
    return RetrofitWrapper.retrofit.create(IRequest::class.java)
}